### ISSUE TITLE
[CoE Starter Kit - BUG] Admin | Sync Template v4 (Security Roles)

### ISSUE THREAD

**Issue by ProductivityCollaboration** [microsoft/coe-starter-kit#8316]
### Does this bug already exist in our backlog?

- [X] I have checked and confirm this is a new bug.

### Describe the issue

There seems to be an issue with retrieving users with an apostrophe in their name. I have 16 errors in the action 
'find the user records - still unknown' and all of them are users with an apostrophe 

`{
  "statusCode": 400,
  "headers": {
    "Cache-Control": "no-cache",
    "x-ms-service-request-id": "bc936a40-97f6-478f-b4e1-b1ce84f2b3a8,e763e63a-a003-4de5-",
    "Set-Cookie": "ARRAffinity=b92acc0c0103a21f64edf2e1913f4ba7c5d97212a99698bbe9291673cb834036fd4ae8d08f3fa44404482caf5a7f8b20e11b185c9c6353130d47c9773598; path=/; secure; HttpOnly,ReqClientId=b5d30ff8-1e60-4b17-87a7-30b6f; expires=Mon, 21-May-2074 05:00:08 GMT; path=/; secure; HttpOnly,ARRAffinity=b92ac4036fd4ae8d08f3fa0d47c9a8627ebc7008DC79571401E165318773598; path=/; secure; HttpOnly",
    "Strict-Transport-Security": "max-age=31536000; includeSubDomains",
    "REQ_ID": "e763e63a-a003-4de5-8988-50988-5072401164f0",
    "CRM.ServiceId": "CRMAppPool",
    "AuthActivityId": "d189a0a869709",
    "x-ms-dop-hint": "48",
    "x-ms-ratelimit-time-remaining-xrm-requests": "1,176.86",
    "x-ms-ratelimit-burst-remaining-xrm-requests": "6914",
    "OData-Version": "4.0",
    "X-Source": "223112158136571882211516918721823",
    "Public": "OPTIONS,GET,HEAD,POST",
    "Date": "Tue, 21 May 2024 05:00:07 GMT",
    "Allow": "OPTIONS,GET,HEAD,POST",
    "Content-Type": "application/json; odata.metadata=full",
    "Expires": "-1",
    "Content-Length": "195"
  },
  "body": {
    "error": {
      "code": "0x80060888",
      "message": "There is an unterminated literal at position 97 in 'admin_name eq 'Pat O'Toole' and _admin_environment_value eq 1ce0dd9e-b000-0000-be0e-8dbbec04b00a'."
    }
  }
}`

there is a very similar issue here - 
https://powerusers.microsoft.com/t5/Building-Power-Apps/There-is-an-unterminated-literal-at-position/td-p/2183167

### Expected Behavior

For the flow to not error if a user has an apostrophe in their name

### What solution are you experiencing the issue with?

Core

### What solution version are you using?

4.24

### What app or flow are you having the issue with?

Sync Template v4 (Security Roles)

### What method are you using to get inventory and telemetry?

Cloud flows

### Steps To Reproduce

Have a user with an apostrophe in their name and run this flow against their account ensuring that it is run through the action of 'Find this users records - still unknown'

**Comment by Jenefer-Monroe** [comment]
Please let me know which action is failing. We have made a recent fix here. A screenshot would be ideal so I dont have to hunt for the action.

**Comment by Jenefer-Monroe** [comment]
Yes you have the same current build as the poster of the other bug. 
Please see https://github.com/microsoft/coe-starter-kit/issues/7884 fixed in April release.
